qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -599,29 +572,23 @@ def _clean_validate_template(cls, md_template, study_id, obj,
             raise QiitaDBDuplicateHeaderError(
                 find_duplicates(md_template.columns))
 
-        # We need to check for some special columns, that are not present on
-        # the database, but depending on the data type are required.
-        missing = cls._check_special_columns(md_template, obj)
+        # Check if we have the columns required for some functionality
+        warning_msg = []
+        for key, restriction in viewitems(restriction_dict):
+            missing = set(restriction.columns).difference(md_template)
 
-        conn_handler = conn_handler if conn_handler else SQLConnectionHandler()
-
-        # Get the required columns from the DB
-        db_cols = get_table_cols(cls._table, conn_handler)
-
-        # Remove the sample_id and study_id columns
-        db_cols.remove('sample_id')
-        db_cols.remove(cls._id_column)
+            if missing:
+                warning_msg.append(
+                    "%s: %s" % (restriction.error_msg, ', '.join(missing)))
 
-        # Retrieve the headers of the metadata template
-        headers = list(md_template.keys())
+        if warning_msg:
+            warnings.warn(
+                "Some functionality will be disabled due to missing "
+                "columns:\n\t%s.\nCheck https://github.com/biocore/qiita/wiki"
```

Comment:
I was also thinking about this, but it's going to be tricky since this
can also be raised from a terminal and you would get an HTML formatted
error message, which is not appropriate. Ideas?

On (Apr-27-15|15:22), Antonio Gonzalez wrote:
> >  
> > -        # Retrieve the headers of the metadata template
> > -        headers = list(md_template.keys())
> > +        if warning_msg:
> > +            warnings.warn(
> > +                "Some functionality will be disabled due to missing "
> > +                "columns:\n\t%s.\nCheck https://github.com/biocore/qiita/wiki"
> 
> Should this be a link so the user can access it?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1075/files#r29197809